pkg/asset/machines/openstack/machines.go

```diff
@@ -108,26 +110,49 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func generateProvider(clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, az string, role, userDataSecret string, trunkSupport bool, rootVolumeAZ string) (*machinev1alpha1.OpenstackProviderSpec, error) {
-	var networks []machinev1alpha1.NetworkParam //nolint:prealloc // declared here for the conditional initialization
+func generateProvider(clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, role, userDataSecret string, trunkSupport bool, failureDomain openstack.FailureDomain) (*machinev1alpha1.OpenstackProviderSpec, error) {
+	var controlPlaneNetwork machinev1alpha1.NetworkParam
+	additionalNetworks := make([]machinev1alpha1.NetworkParam, 0, len(failureDomain.PortTargets)+len(mpool.AdditionalNetworkIDs))
```

Comment:
The final slice of `networkParam` that we put in the providerSpec will be:
```
controlPlaneNetwork + additionalNetworks...
```

I keep them separate until the end just for readability.